gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py

```diff
@@ -0,0 +1,181 @@
+    mt = mt.transmute_rows(hom_count_hetnonref_fix=mt.gt_stats.homozygote_count[1])
+
+    logger.info(
+        "Filtering to rows where homalt hotfix erroneously adjusts het nonref calls..."
+    )
+    mt = mt.filter_rows(mt.hom_count_hotfix != mt.hom_count_hetnonref_fix)
+
+    return mt.rows()
+
+
+def main(args):
+    """
+    Script used to get variants impacted by homalt hotfix.
+    """
+
+    hl.init(log="/get_impacted_variants.log", default_reference="GRCh38")
+
+    logger.info("Reading sparse MT and metadata table with release only samples...")
+    mt = get_gnomad_v3_mt(
+        key_by_locus_and_alleles=True, samples_meta=True, release_only=True,
+    ).select_entries(*SPARSE_ENTRIES)
+
+    if args.test:
+        logger.info("Filtering to two partitions on chr20")
+        mt = hl.filter_intervals(mt, [hl.parse_locus_interval("chr20:1-1000000")])
+        mt = mt._filter_partitions(range(2))
+
+    logger.info(
+        "Adding annotation for whether original genotype (pre-splitting multiallelics) is het nonref..."
+    )
+    # Adding a Boolean for whether a sample had a heterozygous non-reference genotype
+    # Need to add this prior to splitting MT to make sure these genotypes
+    # are not adjusted by the homalt hotfix downstream
+    mt = mt.annotate_entries(_het_non_ref=mt.LGT.is_het_non_ref())
+
+    logger.info("Splitting multiallelics...")
+    mt = hl.experimental.sparse_split_multi(mt, filter_changed_loci=True)
+
+    logger.info("Reading in v3.0 release HT (to get frequency information)...")
+    freq_ht = (
+        release_sites(public=True).versions["3.0"].ht().select_globals().select("freq")
+    )
+
+    logger.info("Reading in info HT...")
+    if file_exists(get_info().path):
+        info_ht = get_info().ht()
+    elif not file_exists(get_info().path) and file_exists(get_info(split=False).path):
+        from gnomad_qc.v3.annotations.generate_qc_annotations import split_info
+
+        info_ht = split_info().drop("old_locus", "old_alleles")
+    else:
+        raise DataException("There is no available split or unsplit info HT for use!")
+
+    logger.info(
+        "Checking for variants with het nonref calls that were incorrectly adjusted with homalt hotfix..."
+    )
+    sites_ht = get_het_non_ref_impacted_var(mt, info_ht, freq_ht)
+
+    logger.info("Densifying MT to het nonref sites only...")
+    # NOTE: densify_sites operates on locus only (doesn't check alleles)
+    # https://github.com/broadinstitute/gnomad_methods/blob/master/gnomad/utils/sparse_mt.py#L102
+    # Thus, it doesn't matter that `sites_ht` has already been split
+    # NOTE: set semi_join_rows to False here because the sites_HT is small
+    mt = densify_sites(
+        mt, sites_ht, last_END_position.versions["3.1"].ht(), semi_join_rows=False,
+    )
+    mt = mt.filter_rows(
+        (hl.len(mt.alleles) > 1) & (hl.is_defined(sites_ht[mt.row_key]))
```

Comment:
do you want to filter lowqual variants here?

Reply:
I thought I needed to, but actually don't since they are not in the freq_ht